### PR TITLE
Use openapi-generator gradle plugin

### PR DESCRIPTION
**Pull request by welandaz**

### CODE REVIEW

**Review by guylabs — CHANGES_REQUESTED** [review]
Nice! Looking wsy more structured.

File: build.gradle.kts [hunk 1]
```diff
@@ -2,118 +2,77 @@ group = "com.gradle.enterprise.api"
...
+            resources.text.fromUri("${baseApiUrl.get()}gradle-enterprise-${gradleEnterpriseVersion}-api.yaml").asFile()
+        })
     )
-}
 
-repositories {
-    mavenCentral()
+openApiGenerate {
+    generatorName.set("java")
+    inputSpec.set(apiSpecificationFile.get().absolutePath)
```
[the remark this answers is not in the document]
Comment (welandaz) [reply build.gradle.kts]: Done

File: build.gradle.kts [hunk 2]
```diff
@@ -2,118 +2,77 @@ group = "com.gradle.enterprise.api"
...
+    outputDir.set("${buildDir}/generated/$name")
```
[the remark this answers is not in the document]
Comment (welandaz) [reply build.gradle.kts]: Done

File: build.gradle.kts [hunk 3]
```diff
@@ -2,118 +2,77 @@ group = "com.gradle.enterprise.api"
...
+    ignoreFileOverride.set("$projectDir/.openapi-generator-ignore")
```
[the remark this answers is not in the document]
Comment (welandaz) [reply build.gradle.kts]: Done

File: build.gradle.kts [hunk 4]
```diff
@@ -2,118 +2,77 @@ group = "com.gradle.enterprise.api"
...
+    modelPackage.set("com.gradle.enterprise.api.model")
```
[the remark this answers is not in the document]
Comment (welandaz) [reply build.gradle.kts]: Done

**Review by guylabs — APPROVED** [review]
All good now 👍

**Review by sorin-florea — APPROVED** [review]

**Pull request merged**